### PR TITLE
Questions-API

### PR DESCRIPTION
**Pull request by graycadeau**
#### What does this PR do?
Has CRUD endpoints for questions

#### PT stories
#160282472
#160282522
#160282529

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -22,3 +24,57 @@ def serialize(self):
...
+
+    def get_all(self):
+
+        """method to fetch all questions"""
+
+        queries = [question.serialize()for question in QUESTIONS]
+
+        return queries
+
+    def serialize(self):
```
Comment (Dhul-Husni) [inline app/models.py]: nice!

File: app/models.py [hunk 2]
```diff
@@ -22,3 +24,57 @@ def serialize(self):
...
+        return {
+            "question": self.question,
+            "id": self.id,
+            "time": self.timestamp
+        }
+
+    def get_specific(self, question_id):
+        """method to fetch a question"""
+
+        for question in QUESTIONS:
```
Comment (Dhul-Husni) [inline app/models.py]: not efficient on large datasets 
please reach out Max as he also has the same issue.

File: app/models.py [hunk 3]
```diff
@@ -22,3 +24,57 @@ def serialize(self):
...
+            if question.id == question_id:
+                return question
+
+        return None
+
+    def delete_specific(self, question_id):
+        """method to delete a question"""
+
+        for question in QUESTIONS:
```
Comment (Dhul-Husni) [inline app/models.py]: same here

**Comment by maxwellgithinji** [comment]
try structuring the data structure in this manner 

questions = {1:{question: value, author:value}}

it is much more efficient to traverse through a dict than a list. tip from @Dhul-Husni , I am working on that issue too

**Pull request merged**